vernac/vernacentries.mli

```diff
@@ -10,7 +10,8 @@
 
 (** Vernac Translation into the Vernac DSL *)
 val translate_vernac
-  : atts:Attributes.vernac_flags
+  : loc:Loc.t
```

Comment:
Use `?loc:Loc.t` or `loc:Loc.t option` [ideally we would use `CAst.t` but this can be done later, `'a located` does also work but it is more or less deprecated in favor of the record]

Reply:
If I pass an option here, then `vernac_command` gets and `?loc` not a `~loc` and I don't like that because it imposes to all clients to test for the option, while we know it is always Some. I prefer to change the type of the grammar entry to generate, at top level, a triple (att,loc,expr). What that be ok?

Reply:
It is not always some, for example when feeding a synthetic term coming from Extern , so this assert would trigger in some use cases that may be done in other tools not in the CI.

If I am not mistaken I don't see the problem of making it an option, in fact if we would like to improve this PR we would use `CAst` and there the location are optional; introducing the clone is gonna be a lot of work but at the core grammar level it won't help due to `Extern`.

What is the burden clients would have concretely? Most location API is already aware of `None` [before it was Loc.dummy]

Reply:
Extern cannot feed "top level" ast nodes. You cannot extend the toplevel non terminal (the one with the attributes and the control flags) but rather the inner one. There you could generate an AST with no loc, but not at the top level.

Anywa, I'll pass an option, I want this merged.

Reply:
Another example of AST without location is what peacoq / eli's new unnamed project do "speculative execution", in this case the IDE does generate some terms that are not in the document and sends them to Coq. Our current API is not good for that, but indeed if we require the location they would either have to use `Loc.dummy` or fault.